src/OVAL/probes/independent/system_info.c

```diff
@@ -383,7 +678,15 @@ int probe_main(probe_ctx *ctx, void *arg)
 	                         "primary_host_name", OVAL_DATATYPE_STRING, hname,
 	                         NULL);
 
-	if (!offline_mode) {
+	if (offline_mode) {
+		if (os_name != unknown)
+			free(os_name);
+		// Don't free os_version, it shares same memory from os_name!
+		if (architecture != unknown)
+			free(architecture);
+		if (hname != unknown)
+			free(hname);
```

Comment:
It's a bit confusing here. 
At first glance it looked like a mistake, because it's comparing pointers, but it is in fact correct.
Getting rid of this free or not free logic by `strdup`ing the `unknown` string would eliminate this confusion.